evm/src/main/java/org/hyperledger/besu/evm/EVM.java

```diff
@@ -203,7 +203,8 @@ public void runToHalt(final MessageFrame frame, final OperationTracer tracing) {
           case 0x50: // POP
             result = PopOperation.staticOperation(frame);
             break;
-          case 0x60: // PUSH1-32
+          case 0x5f: // PUSH0-32
```

Comment:
We need to make sure Push0 is not run pre-Shanghai.  So this optimization needs to go or be guarded, see https://github.com/hyperledger/besu/pull/4633